### PR TITLE
Add usbserial example

### PR DESCRIPTION
**Pull request by dbrgn**
It simply writes the letters 'Rust ' to serial-over-USB output, with a 100ms delay between every two letters.

Depends on https://github.com/japaric/particle-hal/pull/5

### CODE REVIEW

File: examples/usbserial.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+fn setup() {
+    unsafe {
+        usb::USB_USART_Init(BAUD_RATE);
+    }
+}
+
+fn loop_() {
+    unsafe {
+        usb::USB_USART_Send_Data(b'R');
```
Comment (japaric) [inline examples/usbserial.rs]: this can probably be written as a for loop: `for b in b"Rust" { .. }`

Comment (japaric) [reply examples/usbserial.rs]: btw, does the Serial API include a write this byte buffer operation?

**Comment by japaric** [comment]
@homunkulus try

**Comment by homunkulus** [comment]
:hourglass: Trying commit c3ba5f6 with merge 841be9c...

**Comment by homunkulus** [comment]
:sunny: Test successful - [status-travis](https://travis-ci.org/japaric/photon-quickstart/builds/231606136?utm_source=github_status&utm_medium=notification)
State: approved= try=True

File: examples/usbserial.rs (hunk 1 above)
Comment (dbrgn) [reply examples/usbserial.rs]: Doesn't look like it: https://github.com/spark/firmware/blob/develop/hal/inc/usb_hal.h

**Comment by dbrgn** [comment]
Simplified the example.

Writing of entire buffers would probably be something for a higher level abstraction. Where would something like that live, in this crate or in a separate crate?

**Comment by japaric** [comment]
In a crate on top of particle-hal.

Thanks!

@homunkulus r+

**Comment by homunkulus** [comment]
:pushpin: Commit 6430203 has been approved by `japaric`

**Comment by homunkulus** [comment]
:hourglass: Testing commit 6430203 with merge 9499df8...

**Comment by homunkulus** [comment]
:sunny: Test successful - [status-travis](https://travis-ci.org/japaric/photon-quickstart/builds/231615956?utm_source=github_status&utm_medium=notification)
Approved by: japaric
Pushing 9499df8d680a546f15dee7160afe48b7f43b8a29 to master...

**Pull request merged**